src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,134 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+
 public class StringManipulation implements StringManipulationInterface {
+    String string = null;
+    // Default constructor
+    StringManipulation() {} 
+ 
+
+    // Parameterized constructor
+    StringManipulation(String string) {
+        this.string = string;
+    }
 
     @Override
     public String getString() {
-        return null;
+        return this.string;
     }
 
     @Override
     public void setString(String string) {
+        this.string = string;
     }
 
+    // todo: raise nullPointer Exception if string is null
```

Comment:
I noticed this comment you had left for todo.  I thought I would share that this is a great idea for a test case.